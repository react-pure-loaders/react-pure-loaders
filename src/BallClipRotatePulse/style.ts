import { keyframes } from '@emotion/react';
import { balls, globalAnimation } from '../mixins';
import { PRIMARY_COLOR } from '../variables';

const rotate = keyframes({
  '0%': {
    transform: 'rotate(0deg) scale(1)',
  },
  '100%': {
    transform: 'rotate(360deg) scale(1)',
  },
  '50%': {
    transform: 'rotate(180deg) scale(0.6)',
  },
});

const scale = keyframes({
  '100%': {
    transform: 'scale(1)',
  },
  '30%': {
    transform: 'scale(0.3)',
  },
});

export default (color?: string) => ({
  position: 'relative' as 'relative',
  transform: 'translateY(-15px)',

  ' > div': {
    ...balls(color),
    ...globalAnimation(),
    borderRadius: '100%',
    left: '0px',
    position: 'absolute' as 'absolute',
    top: '0px',

    '&:first-of-type': {
      animation: `${scale} 1s 0s cubic-bezier(.09,.57,.49,.9) infinite`,
      background: color || PRIMARY_COLOR,
      height: '16px',
      left: '-7px',
      top: '7px',
      width: '16px',
    },

    '&:last-child': {
      animation: `${rotate} 1s 0s cubic-bezier(.09,.57,.49,.9) infinite`,
      animationDuration: '1s',
      background: 'transparent',
      border: '2px solid',
      borderColor: `${color || PRIMARY_COLOR} transparent ${color
        || PRIMARY_COLOR} transparent`,
      height: '30px',
      left: '-16px',
      position: 'absolute' as 'absolute',
      top: '-2px',
      width: '30px',
    },
  },
});
