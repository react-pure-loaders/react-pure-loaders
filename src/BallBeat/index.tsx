/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

export default function BallBeat({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)}>
                <div/>
                <div/>
                <div/>
            </div>
    );
  }

  return null;
}
