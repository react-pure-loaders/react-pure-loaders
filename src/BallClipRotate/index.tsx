/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

export default function BallClipRotate({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)} data-testid="ball-clip-rotate-container">
                <div/>
            </div>
    );
  }

  return null;
}
