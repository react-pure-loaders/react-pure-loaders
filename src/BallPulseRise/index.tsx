/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

export default function BallPulseRise({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)}>
                <div/>
                <div/>
                <div/>
                <div/>
                <div/>
            </div>
    );
  }

  return null;
}
