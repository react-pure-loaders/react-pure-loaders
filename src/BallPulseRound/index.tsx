/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

export default function BallPulseRound({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)}>
                <div/>
            </div>
    );
  }

  return null;
}
