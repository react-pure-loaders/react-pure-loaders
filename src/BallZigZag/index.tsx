/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

export default function BallZigZag({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)}>
                <div/>
                <div/>
            </div>
    );
  }

  return null;
}
