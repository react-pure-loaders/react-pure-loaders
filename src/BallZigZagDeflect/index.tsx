/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

export default function BallZigZagDeflect({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)}>
                <div/>
                <div/>
            </div>
    );
  }

  return null;
}
