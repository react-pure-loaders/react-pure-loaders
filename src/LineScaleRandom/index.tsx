/** @jsx jsx */
import { jsx } from '@emotion/react';
import style from './style';
import { BaseProps } from '../types';

// eslint-disable-next-line @typescript-eslint/explicit-function-return-type
export default function LineScaleRandom({ color, loading }: BaseProps) {
  if (loading) {
    return (
            <div css={style(color)}>
                <div />
                <div />
                <div />
                <div />
            </div>
    );
  }

  return null;
}
