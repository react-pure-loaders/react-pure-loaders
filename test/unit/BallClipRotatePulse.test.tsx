import React from 'react';
import Chance from 'chance';
import { matchers } from '@emotion/jest';
import { render, cleanup } from '@testing-library/react';


import BallClipRotatePulse from '../../src/BallClipRotatePulse';
import { PRIMARY_COLOR } from '../../src/variables';

expect.extend(matchers);

const chance = new Chance();

describe('<BallClipRotatePulse>', () => {
    afterEach(cleanup);

    test('BallClipRotatePulse should match snapshot', () => {
        const { container } = render(<BallClipRotatePulse loading={true}/>);

        expect(container.firstChild).toMatchSnapshot();
    });

    test.skip('BallClipRotatePulse should have default color', () => {
        const { container } = render(<BallClipRotatePulse loading={true}/>);

        expect(container.firstChild).toHaveStyleRule('background-color', PRIMARY_COLOR, { target: '> div' });
    });

    test.skip('BallClipRotatePulse should have given color', () => {
        const color = chance.color({ format: 'hex' });
        const { container } = render(<BallClipRotatePulse color={color} loading={true}/>);

        expect(container.firstChild).toHaveStyleRule('background-color', color, { target: '> div' });
    });

    test('BallClipRotatePulse should have no children', () => {
        const { container } = render(<BallClipRotatePulse loading={false}/>);

        expect(container.firstChild).toBeNull();
    });

    test('BallClipRotatePulse should have three children', () => {
        const { container } = render(<BallClipRotatePulse loading={true}/>);

        expect(container.querySelectorAll('div')).toHaveLength(3);
    });
});
