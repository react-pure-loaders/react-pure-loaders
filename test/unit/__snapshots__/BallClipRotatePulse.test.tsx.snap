// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BallClipRotatePulse> BallClipRotatePulse should match snapshot 1`] = `
@keyframes animation-0 {
  100% {
    -webkit-transform: scale(1);
    -moz-transform: scale(1);
    -ms-transform: scale(1);
    transform: scale(1);
  }

  30% {
    -webkit-transform: scale(0.3);
    -moz-transform: scale(0.3);
    -ms-transform: scale(0.3);
    transform: scale(0.3);
  }
}

@keyframes animation-1 {
  0% {
    -webkit-transform: rotate(0deg) scale(1);
    -moz-transform: rotate(0deg) scale(1);
    -ms-transform: rotate(0deg) scale(1);
    transform: rotate(0deg) scale(1);
  }

  100% {
    -webkit-transform: rotate(360deg) scale(1);
    -moz-transform: rotate(360deg) scale(1);
    -ms-transform: rotate(360deg) scale(1);
    transform: rotate(360deg) scale(1);
  }

  50% {
    -webkit-transform: rotate(180deg) scale(0.6);
    -moz-transform: rotate(180deg) scale(0.6);
    -ms-transform: rotate(180deg) scale(0.6);
    transform: rotate(180deg) scale(0.6);
  }
}

.emotion-0 {
  position: relative;
  -webkit-transform: translateY(-15px);
  -moz-transform: translateY(-15px);
  -ms-transform: translateY(-15px);
  transform: translateY(-15px);
}

.emotion-0>div {
  border-radius: 100%;
  height: 15px;
  margin: 2px;
  width: 15px;
  background-color: #ffffff;
  -webkit-animation-fill-mode: both;
  animation-fill-mode: both;
  left: 0px;
  position: absolute;
  top: 0px;
}

.emotion-0>div:first-of-type {
  -webkit-animation: animation-0 1s 0s cubic-bezier(.09,.57,.49,.9) infinite;
  animation: animation-0 1s 0s cubic-bezier(.09,.57,.49,.9) infinite;
  background: #ffffff;
  height: 16px;
  left: -7px;
  top: 7px;
  width: 16px;
}

.emotion-0>div:last-child {
  -webkit-animation: animation-1 1s 0s cubic-bezier(.09,.57,.49,.9) infinite;
  animation: animation-1 1s 0s cubic-bezier(.09,.57,.49,.9) infinite;
  -webkit-animation-duration: 1s;
  animation-duration: 1s;
  background: transparent;
  border: 2px solid;
  border-color: #ffffff transparent #ffffff transparent;
  height: 30px;
  left: -16px;
  position: absolute;
  top: -2px;
  width: 30px;
}

<div
  class="emotion-0"
>
  <div />
  <div />
</div>
`;
